app/models/server/models/LivechatDepartmentAgents.js

```diff
@@ -65,10 +65,14 @@ export class LivechatDepartmentAgents extends Base {
 
 		const onlineUsernames = _.pluck(onlineUsers.fetch(), 'username');
 
+		// get fully booked agents, to ignore them from the query
+		const currentUnavailableAgents = Promise.await(Users.getUnavailableAgents(departmentId, extraQuery)).map((u) => u.username);
```

Comment:
```suggestion
		const currentUnavailableAgents = Users.getUnavailableAgents(departmentId, extraQuery).map((u) => u.username);
```

The `getUnavailableAgents` method isn't async so I don't think we need to resolve a promise here

Reply:
afaik, `rawCollection` queries are not handled by meteor, so you always need to wait for them 😬 even when they're on a Meteor Model.

Reply:
Thanks man!